packages/evo/src/wrap/js-engine/Date.ts

```diff
@@ -0,0 +1,203 @@
+// language=javascript
+export const dateShim = `class Date {
+
+  _timestamp;
+
+  constructor(...args) {
+    if (args.length === 0) {
+      this._timestamp = Date.now();
+    } else if (args.length > 1) {
+      this._timestamp = Date.UTC(...args);
+    } else if (typeof args[0] === "string") {
+      this._timestamp = Date.parse(args[0]);
+    } else {
+      this._timestamp = args[0].timestamp;
+    }
+  }
+
+  static now() {
+    return __wrap_subinvoke("plugin/datetime", "now", clean(undefined)).value;
```

Comment:
This will return the object but not an instance of the Date class (like how it would in standard JS)

Reply:
I'm a bit confused by this, would you mind explaining a bit more?

Reply:
The Mozilla API reference suggests `Date.now` should return a number: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/now

Reply:
I wasn't sure how to test the shim so maybe it's not returning what I think it is

Reply:
Nvm, I thought it returned the Date object instead of the number.